### DISCUSSION TITLE
How can I pass prisma user to session without querying user it on every session?

### DISCUSSION THREAD

**Discussion by sbkobaidze** [nextauthjs/next-auth#9222] [Help] [answered]
I think  I should pass user from adapter directly, but not sure how that will work - does anybody know how to that

`` 
export const authOptions: NextAuthOptions = {
  // Configure one or more authentication providers
  adapter: PrismaAdapter(prisma),
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_ID!,
      clientSecret: process.env.GOOGLE_SECRET!,
    }),
    DiscordProvider({
      clientId: process.env.DISCORD_CLIENT_ID!,
      clientSecret: process.env.DISCORD_CLIENT_SECRET!
    })
    // ...add more providers here
  ],

  callbacks: {
    session: async (session, user) => {

      console.log(user,'test1234')

    },
  }, ``

**Comment by lvanoverberghe** [comment]
You can augment the session object with anything from the user:  

      session: async ({ session, user }) => {
      session.user = {
        ...session.user,
        id: user.id,
      };
      return session;
    },

**Reply by sbkobaidze** [reply]
Hey, thanks for your response. The thing is that when I console.log(user) in session callback it is undefined, looks like I have to pass the user first from jwt callback but when I do that I am having problem with Google OAuth

**Reply by lvanoverberghe** [reply]
Can you enable debug mode so you can see more?

**Reply by sbkobaidze** [reply]
Yes looks like jwt cookie data exceeds 4096 bytes

**Reply by lvanoverberghe** [reply, marked as answer]
You should only store what you need in the jwt cookie, you need to change the jwt callback.
